EventListener/ReportSubscriber.php

```diff
@@ -0,0 +1,268 @@
+    {
+        return [
+            ReportEvents::REPORT_ON_BUILD    => ['onReportBuilder', 0],
+            ReportEvents::REPORT_ON_GENERATE => ['onReportGenerate', 0],
+        ];
+    }
+
+    public function onReportBuilder(ReportBuilderEvent $event): void
+    {
+        if (!$event->checkContext([self::CONTEXT_COMPANY_SEGMENTS])) {
+            return;
+        }
+
+        $columns = $this->companyReportData->getCompanyData();
+        unset($columns['companies_lead.is_primary'], $columns['companies_lead.date_added']);
+
+        $segmentList = $this->getFilterSegments();
+
+        $segmentFilter = [self::COMPANY_SEGMENTS_XREF_PREFIX.'.segment_id' => [
+            'alias'     => 'companysegments',
+            'label'     => 'mautic.company_segments.report.company_segments',
+            'type'      => 'select',
+            'list'      => $segmentList,
+            'operators' => [
+                'in'       => 'mautic.core.operator.in',
+                'notIn'    => 'mautic.core.operator.notin',
+                'empty'    => 'mautic.core.operator.isempty',
+                'notEmpty' => 'mautic.core.operator.isnotempty',
+            ],
+        ],
+        ];
+        $filters = array_merge($columns, $segmentFilter);
+
+        $event->addTable(
+            self::CONTEXT_COMPANY_SEGMENTS,
+            [
+                'display_name' => 'mautic.company_segments.report.company_segments',
+                'columns'      => $columns,
+                'filters'      => $filters,
+            ],
+            'companies'
+        );
+
+        $event->addGraph(self::CONTEXT_COMPANY_SEGMENTS, 'line', 'mautic.lead.graph.line.companies');
+        $event->addGraph(self::CONTEXT_COMPANY_SEGMENTS, 'pie', 'mautic.lead.graph.pie.companies.industry');
+        $event->addGraph(self::CONTEXT_COMPANY_SEGMENTS, 'pie', 'mautic.lead.table.pie.company.country');
+        $event->addGraph(self::CONTEXT_COMPANY_SEGMENTS, 'table', 'mautic.lead.company.table.top.cities');
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public function getFilterSegments(): array
+    {
+        $segments    = $this->companySegmentRepository->getSegmentObjectsViaListOfIDs([]);
+        $segmentList = [];
+        foreach ($segments as $segment) {
+            $segmentList[(string) $segment->getId()] = (string) $segment->getName();
+        }
+
+        return $segmentList;
+    }
+
+    public function onReportGenerate(ReportGeneratorEvent $event): void
+    {
+        if (!$event->checkContext([self::CONTEXT_COMPANY_SEGMENTS])) {
+            return;
+        }
+
+        $qb       = $event->getQueryBuilder();
+        $filters  = $event->getReport()->getFilters();
+        $options  = $event->getOptions()['columns'];
+        $orGroups = [];
+        $andGroup = [];
+
+        $qb
+            ->from(MAUTIC_TABLE_PREFIX.self::COMPANY_TABLE, self::COMPANIES_PREFIX);
+
+        $expr     = $qb->expr();
+
+        if (count($filters)) {
```

Comment:
It seems you did not execute local phpstan and rector. Sure they are not in the CI, but they are providing a better code quality check, than ones in Mautic.

Reply:
I actually used phpstan, rector and fixcs before commit and push. If it's about the spaces between $expr and =, that's something that csfixer did

Reply:
Hm, running PHPStan locally with `composer phpstan` in the directory with the code i can see PHPStan reports 20 errors.

Reply:
Done